contracts/src/registry/ETHRegistrar.sol

```diff
@@ -0,0 +1,214 @@
+     */
+    function valid(string memory name) public pure returns (bool) {
+        return bytes(name).length >= 3;
+    }
+
+    /**
+     * @dev Check if a name is available.
+     * @param name The name to check.
+     * @return True if the name is available, false otherwise.
+     */
+    function available(string calldata name) external view returns (bool) {
+        address subregistry = address(registry.getSubregistry(name));
+        return subregistry == address(0);
+    }
+
+
+    /**
+     * @dev Get the price to register or renew a name.
+     * @param name The name to get the price for.
+     * @param duration The duration of the registration or renewal.
+     * @return price The price to register or renew the name.
+     */ 
+    function rentPrice(string memory name, uint256 duration) public view override returns (IPriceOracle.Price memory price) {
+        (uint96 expiry, ) = registry.nameData(NameUtils.labelToTokenId(name));
+        price = prices.price(name, uint256(expiry), duration);
+    }    
+
+
+    /**
+     * @dev Make a commitment for a name.
+     * @param name The name to commit.
+     * @param owner The address of the owner of the name.
+     * @param secret The secret of the name.
+     * @param subregistry The registry to use for the commitment.
+     * @param flags The flags to use for the commitment.
+     * @param duration The duration of the commitment.
+     * @return The commitment.
+     */
+    function makeCommitment(
+        string memory name,
+        address owner,
+        bytes32 secret,
+        address subregistry,
+        uint96 flags,
+        uint64 duration
+    ) public pure override returns (bytes32) {        
+        return
+            keccak256(
+                abi.encode(
+                    name,
+                    owner,
+                    secret,
+                    subregistry,
+                    flags,
+                    duration
+                )
+            );
+    }
+
+
+    /**
+     * @dev Commit a commitment.
+     * @param commitment The commitment to commit.
+     */
+    function commit(bytes32 commitment) public override {
+        if (commitments[commitment] + maxCommitmentAge >= block.timestamp) {
+            revert UnexpiredCommitmentExists(commitment);
+        }
+        commitments[commitment] = block.timestamp;
+    }
+
+
+
+    /**
+     * @dev Register a name.
+     * @param name The name to register.
+     * @param owner The owner of the name.
+     * @param subregistry The subregistry to register the name in.
+     * @param flags The flags to set on the name.   
+     * @param duration The duration of the registration.
+     * @return tokenId The token ID of the registered name.
+     */
+    function register(
+        string calldata name,
+        address owner,
+        IRegistry subregistry,
+        uint96 flags,
+        uint64 duration
+    ) external payable onlyRole(CONTROLLER_ROLE) returns (uint256 tokenId) {
```

Comment:
```suggestion
    ) external payable returns (uint256 tokenId) {
```
I think we want this to be publicly callable now, right?

Reply:
If it's publicly callable what's the purpose of the controler role then?

Reply:
There isn't one - the registry needs a controller, which will be this contract. This contract doesn't need a controller role because it's intended to be called by end users.